### PR TITLE
[DOCS] Add runtime fields to Painless Guide

### PR DESCRIPTION
**Pull request by lockewritesdocs**
This PR:
* Adds a new page in the Painless guide for [using runtime fields](https://elasticsearch_67781.docs-preview.app.elstc.co/guide/en/elasticsearch/painless/master/painless-runtime-fields.html)
* Adds a new page for [runtime fields context](https://elasticsearch_67781.docs-preview.app.elstc.co/guide/en/elasticsearch/painless/master/painless-runtime-fields-context.html) in Painless

### CODE REVIEW

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc [hunk 1]
```diff
@@ -0,0 +1,218 @@
+[[painless-runtime-fields-context]]
+=== Runtime fields context
+beta::[]
+Use a Painless script to return a calculated value for a
```
Comment (javanna) [inline docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:4]: to calculate and emit runtime field values?

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+<<painless-runtime-fields,runtime field>>, defined either in the mappings or as
+part of a query. You can also use runtime fields to override values for indexed
+fields.
```
Comment (javanna) [inline docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:7]: I would say that the link is enough and we don't need to expand on where they are defined and that they can override existing fields?

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+that runtime fields require when defining a Painless script.
+
+Because `day_of_week` is a runtime field, it isn't indexed, and the included
+script only runs at query time:
+
+[source,console]
+----
+PUT seats/_mapping
+{
+    "dynamic": "runtime",
```
Comment (javanna) [inline docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:133]: is dynamic:runtime needed here?

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+definition:
+
+[source,console]
+----
+GET seats/_search
+{
+  "size": 2,
+  "fields": [
+    "time",
+    "day_of_week"
```
Comment (javanna) [inline docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:158]: I wonder if we should do something more meaningful like querying the field or aggregating on it, rather than only retrieving it.

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    ]
+  }
+}
+----
+// TESTRESPONSE[s/\.\.\./"took" : $body.took,"timed_out" : $body.timed_out,"_shards" : $body._shards,/]
+// TESTRESPONSE[s/"value" : "11"/"value": $body.hits.total.value/]
+
+This is just one example of how you can use runtime fields. You can include
+Painless scripts in the context of runtime fields to override values for
+indexed fields or create fields that exist only as part of the query.
```
Comment (javanna) [inline docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:215]: I wonder if these details are needed here as we already link to the runtime fields docs.

File: docs/reference/mapping/runtime.asciidoc [hunk 6]
```diff
@@ -1,5 +1,7 @@
 [[runtime]]
 == Runtime fields
+beta::[]
```
Comment (javanna) [inline docs/reference/mapping/runtime.asciidoc:3]: can we make the beta change separately given that it is not related to this specific PR?

**Review by javanna — COMMENTED** [review]
left a couple of comments, thanks!

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+
+*Example*
+
+Let's break down each of the steps from the
+<<painless-context-examples,context examples>> to understand how runtime fields
+work within the context of Painless.
+
+[[painless-runtime-fields-mappings]]
+==== Create the index mappings
```
Comment (jdconrad) [inline docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:40]: I'm worried that having the full example setup as part of this page is confusing since the current setup for all the other contexts requires setup from the original page with a link there. Would the longer term plan be to do this for all the context pages so there's no multi-page requirement?

**Review by jdconrad — APPROVED** [review]
LGTM! I left one question as a comment. Thanks for working on this.

File: docs/reference/mapping/runtime.asciidoc (hunk 6 above)
Comment (lockewritesdocs) [reply docs/reference/mapping/runtime.asciidoc:3]: Yes -- I can remove the beta admonitions and include them in a separate PR.

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc (hunk 7 above)
Comment (lockewritesdocs) [reply docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:40]: The main reason I included the examples is that getting started with the current [context examples](https://www.elastic.co/guide/en/elasticsearch/painless/7.11/painless-context-examples.html) puts more work on the user. They have to download a file, link off to other pages, and use both curl requests and Console examples to get things working. I wanted to streamline that process so that users can set up this entire example without leaving the page and navigating around.

Going forward, we can either improve the context examples page or integrate the examples directly into each context page. There are pros and cons to both approaches, and I'd like to chat with you further about which direction to take.

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc (hunk 2 above)
Comment (lockewritesdocs) [reply docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:7]: Works for me 👍

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc (hunk 3 above)
Comment (lockewritesdocs) [reply docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:133]: No 😄 Good catch -- I'll remove it.

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc (hunk 4 above)
Comment (lockewritesdocs) [reply docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:158]: @javanna, makes sense to me! How about something like this? Totally open to better options:
```
GET seats/_search
{
  "size": 2,
  "fields": [
    "time",
    "day_of_week"
    ],
    "aggs": {
      "day_of_week": {
        "terms": {
          "field": "day_of_week",
          "size": 10
        }
      }
    }
}
```

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc (hunk 5 above)
Comment (lockewritesdocs) [reply docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:215]: I'm ok with removing this bit.

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc (hunk 7 above)
Comment (jdconrad) [reply docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:40]: My initial reaction is let's just integrate the examples into each page as that seems like it's simplest for the user and get rid of any downloaded files. I thought at the time it might be useful to have a larger data set for aggregations examples, but that's proven to not be the case.

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc (hunk 4 above)
Comment (javanna) [reply docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:158]: yea looks good to me, you could even make it shorter and use size: 0 as well as remove the fields parameter.

Comment (lockewritesdocs) [reply docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:158]: Oh nice! I'll do that so we can focus on the aggregation more easily.

File: docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc (hunk 7 above)
Comment (lockewritesdocs) [reply docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc:40]: I have some ideas about how we can approach the examples. Let's chat about the different ways we can solve this problem and do that work in a separate PR.
